scheduler/src/cook/mesos/util.clj

```diff
@@ -36,13 +85,16 @@
        (map first)))
 
 (defn categorize-job
-  "Return the category of the job. Currently jobs can be :normal or :gpu. This
-   is used to give separate queues for scarce & non-scarce resources"
-  [job]
-  (let [resources (:job/resource job)]
-    (if (some #(= :resource.type/gpus (:resource/type %)) resources)
-      :gpu
-      :normal)))
+      "Return the category of the job. Currently jobs can be :normal or :gpu. This
+       is used to give separate queues for scarce & non-scarce resources"
+      [job]
+      (let [categorize-job-miss
+            (fn [job]
+                (let [resources (:job/resource job)]
+                     (if (some #(= :resource.type/gpus (:resource/type %)) resources)
+                       :gpu
+                       :normal)))]
+           (lookup-cache-datomic-entity! categorize-job-cache categorize-job-miss job)))
```

Comment:
Is `job` a Datomic entity?  If so, it has a builtin reference to `db` and we don't want to cache that.  @pschorf removed a huge memory leak before related to this.

Reply:
If it's not a Datomic entity, we should rename the function.

Reply:
Yes, it is a datomic entity. But that's OK. lookup-cache-datomic-entity! is a specialized version of lookup-cache-generic! that can use datomic entities as cache keys without the memory leak.